pkg/common/utils/utils_test.go

```diff
@@ -331,3 +331,39 @@ func TestGetApplicationIDFromPod(t *testing.T) {
 		})
 	}
 }
+
+func TestMergeMaps(t *testing.T) {
+	result := MergeMaps(nil, nil)
+	assert.Assert(t, result == nil)
+
+	result = MergeMaps(nil, map[string]string{"a": "b"})
+	assert.Assert(t, result != nil)
+	assert.Equal(t, len(result), 1)
+	assert.Equal(t, result["a"], "b")
+
+	result = MergeMaps(map[string]string{"a": "b"}, nil)
+	assert.Assert(t, result != nil)
+	assert.Equal(t, len(result), 1)
+	assert.Equal(t, result["a"], "b")
+
+	result = MergeMaps(map[string]string{"a": "a1"}, map[string]string{"a": "a2"})
+	assert.Assert(t, result != nil)
+	assert.Equal(t, len(result), 1)
+	assert.Equal(t, result["a"], "a2")
+
+	result = MergeMaps(map[string]string{
+		"a": "a1",
+		"b": "b1",
+		"c": "c1",
+	}, map[string]string{
+		"a": "a2",
+		"b": "b2",
+		"d": "d2",
+	})
+	assert.Assert(t, result != nil)
+	assert.Equal(t, len(result), 4)
+	assert.Equal(t, result["a"], "a2")
+	assert.Equal(t, result["b"], "b2")
+	assert.Equal(t, result["c"], "c1")
+	assert.Equal(t, result["d"], "d2")
+}
```

Comment:
nit: if it is possible in go it is suggested to write table driven tests. It is more clear and elegant.
```
testCases := []struct {
		name string
		map1       map[string]string
		map2 map[string]string
		expectedResult map[string]string
	}{
		{"Nil maps", nil, nil, nil},
                  ...... <create here the other cases>
	}
	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			assert.DeepEqual(t, MergeMaps(tc.map1, tc.map2), tc.expectedResult)
		})
	}
```